src/serialize.ts

```diff
@@ -97,7 +97,13 @@ const Serializer = /*@__PURE__*/ (function () {
         return this.serializeBuiltInType(objName, object);
       }
       if (typeof object.toJSON === "function") {
-        return objName + this.$object(object.toJSON());
+        const json = object.toJSON();
+        return (
+          objName +
+          (typeof json === "object" && json !== null
```

Comment:
(micro optimization: when value is null, we don't need to check it's type)

```suggestion
          (json !== null && typeof json === "object"
```